fix(image_rs): resolve hardlinks with absolute paths during unpack

Park-Jiyeonn opened the pull request:
### Problem
[#1148](https://github.com/confidential-containers/guest-components/issues/1148)

When unpacking container images using `coco` in guest-pull mode, hardlinks with absolute `Linkname`s (e.g., `/usr/bin/unzip`) cause unpack failures due to canonicalization errors.

### Solution
If `file.unpack_in` fails, it attempts a fallback using `try_hardlink_fallback`, which creates hard links for entries with absolute link paths.

The fallback logic resolves the destination path, normalizes the link path to ensure safety (e.g., removes root prefix and prevents escaping above root). If the fallback succeeds, the unpack continues; otherwise, it returns an error.

### Notes
If you add `println!("{:?}", e)` inside the `match file.unpack_in(destination).await {Err(e) => {} }` block while running the `test_unpack` test case, you can see the output which similar to the issue:
```log
Custom { kind: NotFound, error: TarError { desc: "failed to unpack `/tmp/image_test_dir/usr/bin/test_binary_hardlink`", io: Custom { kind: NotFound, error: "No such file or directory (os error 2) while canonicalizing /usr/bin/test_binary" } } }
```

Xynnn007 reviewed (commented):
Thanks for the fix! There are some comments inside the code.

Park-Jiyeonn commented on image-rs/src/stream/unpack.rs:
```diff
@@ -364,16 +369,85 @@ async fn set_perms_ownerships(
+                if !out.pop() {
+                    return None;
+                }
+            }
```
It's not necessarily to normalize, how about this:

only remove the prefix '/' from the absolute linkname and spell it into `destination`, then use canonization to resolve the real path and apply `starts_with (destination)` to prevent escape.

Park-Jiyeonn commented:
> you may want to add a negative test case to make sure the anti-escape code works

Thx! I've added a negative test case to cover the anti-escape logic, 
```rust
+    #[tokio::test]
+    async fn test_unpack_rejects_escaping_absolute_hardlink() {
+        let td = tempfile::tempdir().unwrap();
......
```

and updated the error message to:
```rust
-                            format!("Fallback failed {f} after original unpack error {e}"),
+                            format!("Fallback failed, {f:?}, after original unpack error: {e:?}"),
```
